processor/resourcedetectionprocessor/internal/azure/aks/aks.go

```diff
@@ -53,6 +55,14 @@ func (d *Detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 	}
 	if d.resourceAttributes.CloudPlatform.Enabled {
 		attrs.PutStr(conventions.AttributeCloudPlatform, conventions.AttributeCloudPlatformAzureAKS)
+
+		if d.resourceAttributes.K8sClusterName.Enabled {
```

Comment:
k8s cluster name should still be recorded when cloud platform is disabled, so I think this should be unnested.